src/containerapp/azext_containerapp/_clients.py

```diff
@@ -548,13 +541,17 @@ def update(cls, cmd, resource_group_name, name, managed_environment_envelope, no
 
         if no_wait:
             return r.json()
-        elif r.status_code == 201:
+        elif r.status_code == 202:
             operation_url = r.headers.get(HEADER_LOCATION)
-            response = poll_results(cmd, operation_url)
-            if response is None:
-                raise ResourceNotFoundError("Could not find a managed environment")
+            if "managedEnvironmentOperationStatuses" in operation_url:
+                poll_status(cmd, operation_url)
+                r = send_raw_request(cmd.cli_ctx, "GET", request_url)
             else:
-                return response
+                response = poll_results(cmd, operation_url)
```

Comment:
It would be better to make sure it is  operation result uri when polling results